addons/website_forum/models/forum_post.py

```diff
@@ -803,13 +806,15 @@ def _search_get_detail(self, website, order, options):
             'name': {'name': 'name', 'type': 'text', 'match': True},
             'website_url': {'name': 'website_url', 'type': 'text', 'truncate': False},
         }
-
+        values = options.get('values') or False
         domain = website.website_domain()
-        domain = expression.AND([domain, [('state', '=', 'active'), ('can_view', '=', True)]])
+        domain = expression.AND([domain, [('can_view', '=', True)]])
         include_answers = options.get('include_answers', False)
         if not include_answers:
             domain = expression.AND([domain, [('parent_id', '=', False)]])
         forum = options.get('forum')
+        if values is False or values.get('user').karma < values.get('forum').karma_moderate:
```

Comment:
why using `.get` and then `.karma` ?

if it's none it will crach anyway, so you can do
```suggestion
        if not values or values['user'].karma < values['forum'].karma_moderate: